src/dist/replication/lib/replica_bulk_load.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "replica.h"
+#include "replica_stub.h"
+
+#include <fstream>
+
+#include <dsn/dist/block_service.h>
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/dist/replication/replication_app_base.h>
+#include <dsn/utility/fail_point.h>
+#include <dsn/utility/filesystem.h>
+
+namespace dsn {
+namespace replication {
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica::on_bulk_load(const bulk_load_request &request, /*out*/ bulk_load_response &response)
+{
+    _checker.only_one_thread_access();
+
+    response.pid = request.pid;
+    response.app_name = request.app_name;
+    response.err = ERR_OK;
+
+    if (status() != partition_status::PS_PRIMARY) {
+        dwarn_replica("receive bulk load request with wrong status {}", enum_to_string(status()));
+        response.err = ERR_INVALID_STATE;
+        return;
+    }
+
+    if (request.ballot != get_ballot()) {
+        dwarn_replica(
+            "receive bulk load request with wrong version, remote ballot={}, local ballot={}",
+            request.ballot,
+            get_ballot());
+        response.err = ERR_INVALID_STATE;
+        return;
+    }
+
+    ddebug_replica(
+        "receive bulk load request, remote provider = {}, cluster_name = {}, app_name = {}, "
+        "meta_bulk_load_status = {}, local bulk_load_status = {}",
+        request.remote_provider_name,
+        request.cluster_name,
+        request.app_name,
+        enum_to_string(request.meta_bulk_load_status),
+        enum_to_string(get_bulk_load_status()));
+
+    error_code ec = do_bulk_load(request.app_name,
+                                 request.meta_bulk_load_status,
+                                 request.cluster_name,
+                                 request.remote_provider_name);
+    if (ec != ERR_OK) {
+        response.err = ec;
+        response.primary_bulk_load_status = get_bulk_load_status();
+        return;
+    }
+
+    report_bulk_load_states_to_meta(
+        request.meta_bulk_load_status, request.query_bulk_load_metadata, response);
+    if (response.err != ERR_OK) {
+        return;
```

Comment:
Add some log in failure case for `do_bulk_load ` and `report_bulk_load_states_to_meta ` would be more helpful for debuging and trouble shooting.

Reply:
There will be logs in `do_bulk_load` and `report_bulk_load_states_to_meta`, so I don't add logs in `on_bulk_load` repeatedly.